### PR TITLE
[Keystone] Minor bugfixes

### PR DESCRIPTION
**Pull request by bolekk**
1. Fix workflowName decoding in KeystoneFeedsConsumer.sol
2. Fix potential panic in triggerSubscriber.UnregisterTrigger() (KS-225) and make (un)register idempotent
3. Fix empty report check in write_target
4. Some comment/log improvements

### CODE REVIEW

File: core/capabilities/remote/trigger_subscriber.go [hunk 1]
```diff
@@ -149,7 +158,10 @@ func (s *triggerSubscriber) UnregisterTrigger(ctx context.Context, request commo
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
-	close(s.registeredWorkflows[request.Metadata.WorkflowID].callback)
+	state := s.registeredWorkflows[request.Metadata.WorkflowID]
+	if state != nil && state.callback != nil {
+		close(state.callback)
+	}
 	delete(s.registeredWorkflows, request.Metadata.WorkflowID)
```
Comment (krehermann) [inline core/capabilities/remote/trigger_subscriber.go:165]: should the delete be guarded too?

state, exists := s.registeredWorkflows...
if state != nil ..
if exists {
delete()
}

Comment (bolekk) [reply core/capabilities/remote/trigger_subscriber.go:165]: It shouldn't matter for delete() right?

Comment (krehermann) [reply core/capabilities/remote/trigger_subscriber.go:165]: no, it's no strictly necessary. you can ignore

**Review by krehermann — COMMENTED** [review]
golang looks good, not sure about the solidity. why isn't there a test for it?

File: contracts/src/v0.8/keystone/KeystoneFeedsConsumer.sol [hunk 2]
```diff
@@ -88,8 +88,8 @@ contract KeystoneFeedsConsumer is IReceiver, ConfirmedOwner {
     // workflow_owner           // offset 74, size 20
     // report_name              // offset 94, size  2
     assembly {
-      // shift right by 22 bytes to get the actual value
-      workflowName := shr(mul(22, 8), mload(add(metadata, 64)))
+      // no shifting needed for bytes10 type
+      workflowName := mload(add(metadata, 64))
```
Comment (archseer) [inline contracts/src/v0.8/keystone/KeystoneFeedsConsumer.sol:92]: Ah right, I fixed this in the forwarder but didn't realize there's a matching snippet in the consumer

**Review by krehermann — APPROVED** [review]

**Review by archseer — APPROVED** [review]

**Pull request merged**